### ISSUE TITLE
enableLocationRequest() opens multiple settings dialogs in Android

### ISSUE THREAD

**Issue by spbond** [NativeScript/nativescript-geolocation#29]
I am using API version 23.

When calling enableLocationRequest(), if the user denies permissions one or more times before finally accepting, it appears to result in a call to the activityRequestPermissionsEvent handler once for every time the enableLocationRequest() was called, each time it is called.  If the location service settings were turned off, this results in the user having to back out of the location services setting screen multiple times before getting back to the app.

#### Console output:
>JS: Request 0:
>JS: requestCode: 5000 permissions: [Ljava.lang.String;@5735651 grantResults: [I@a0e9eb6
>JS: permission not granted!!!
>JS: Unhandled Promise rejection: Location service is not enabled or using it is not granted. ; Zone: <root> ; Task:  ; Value: Error: Location service is not enabled or using it is not granted.
>JS: Error: Uncaught (in promise): Error: Location service is not enabled or using it is not granted.
>JS: Request 1:
>JS: requestCode: 5000 permissions: [Ljava.lang.String;@8af5590 grantResults: [I@c197e89
>JS: permission not granted!!!
>JS: requestCode: 5000 permissions: [Ljava.lang.String;@8af5590 grantResults: [I@c197e89
>JS: permission not granted!!!
>JS: Unhandled Promise rejection: Location service is not enabled or using it is not granted. ; Zone: <root> ; Task:  ; Value: Error: Location service is not enabled or using it is not granted.
>JS: Error: Uncaught (in promise): Error: Location service is not enabled or using it is not granted.
>JS: Request 2:
>JS: requestCode: 5000 permissions: [Ljava.lang.String;@9a7becb grantResults: [I@49046a8
>JS: permission not granted!!!
>JS: requestCode: 5000 permissions: [Ljava.lang.String;@9a7becb grantResults: [I@49046a8
>JS: permission not granted!!!
>JS: requestCode: 5000 permissions: [Ljava.lang.String;@9a7becb grantResults: [I@49046a8
>JS: permission not granted!!!
>JS: Unhandled Promise rejection: Location service is not enabled or using it is not granted. ; Zone: <root> ; Task:  ; Value: Error: Location service is not enabled or using it is not granted.
>JS: Error: Uncaught (in promise): Error: Location service is not enabled or using it is not granted.
>JS: Request 3:
>JS: requestCode: 5000 permissions: [Ljava.lang.String;@ad6edf2 grantResults: [I@52f3c43
>JS: permission not granted!!!
>JS: requestCode: 5000 permissions: [Ljava.lang.String;@ad6edf2 grantResults: [I@52f3c43
>JS: permission not granted!!!
>JS: requestCode: 5000 permissions: [Ljava.lang.String;@ad6edf2 grantResults: [I@52f3c43
>JS: permission not granted!!!
>JS: requestCode: 5000 permissions: [Ljava.lang.String;@ad6edf2 grantResults: [I@52f3c43
>JS: permission not granted!!!
>JS: Unhandled Promise rejection: Location service is not enabled or using it is not granted. ; Zone: <root> ; Task:  ; Value: Error: Location service is not enabled or using it is not granted.
>JS: Error: Uncaught (in promise): Error: Location service is not enabled or using it is not granted.

As you can see, the fourth request results in the callback being executed 4 times. I was able to solve this issue for myself by modifying the code in the enableLocationRequestCore() method in application.android.js starting at line 129 to the following:
```javascript
        var requestPermissionsResultHandler_1 = function (data) {
            appModule.android.off(appModule.AndroidApplication.activityRequestPermissionsEvent, requestPermissionsResultHandler_1);
            console.log('requestCode: ' + data.requestCode + ' permissions: ' + data.permissions + ' grantResults: ' + data.grantResults);
            if (data.requestCode === 5000) {
                if (data.grantResults.length > 0 && data.grantResults[0] == android.content.pm.PackageManager.PERMISSION_GRANTED) {
                    console.log("permission granted!!!");
                    enableLocationServiceRequest(currentContext, successCallback, successArgs, errorCallback, errorArgs);
                }
                else {
                    console.log("permission not granted!!!");
                    if (errorCallback) {
                        errorCallback.apply(_this, errorArgs);
                    }
                }
            }
        };
        appModule.android.on(appModule.AndroidApplication.activityRequestPermissionsEvent, requestPermissionsResultHandler_1);
````
This unbinds the activityRequestPermissionsEvent handler when it is called, in the same manner you do in the enableLocationServiceRequest() function in the same file.

**Comment by bradmartin** [comment]
I've reproduced this on 23 also.

**Comment by bradmartin** [comment]
@valentinstoychev @enchev any chance we can get this fix and update published? It comes off amateurish with the multiple dialogs opening. If I get the time I'll gladly submit the PR but IDK when that'll happen right now.

**Comment by nsndeck** [comment]
Issue fixed and 0.0.14 version is already published to npm.

**Comment by bradmartin** [comment]
Awesome.
